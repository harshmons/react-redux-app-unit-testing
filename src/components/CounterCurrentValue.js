import React from "react";
import "../styles/App.css";
const CounterCurrentvalue = (props) =>{
    return(
        <div className="currentValueContainer">
            Counter current value : <strong> {props.counterValue} </strong>
        </div>    
    )
}

export default CounterCurrentvalue;