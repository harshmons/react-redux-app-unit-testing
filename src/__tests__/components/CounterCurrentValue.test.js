import React from 'react';
import CounterCurrentValue from "../../components/CounterCurrentValue"
import { shallow } from "enzyme";

const enzymeWrapper = shallow(<CounterCurrentValue />);

describe("<CounterCurrentValue />", () => {
  test("should render passed props counterValue", () => {
    const props = {
        counterValue:10
    }
    enzymeWrapper.setProps({
        ...props
    })
    expect(enzymeWrapper.text()).toBe("Counter current value :  " + props.counterValue + " ");
  });

  test("should matches the stored snapshot",()=>{
    expect(enzymeWrapper.text()).toMatchSnapshot();
  })
});
