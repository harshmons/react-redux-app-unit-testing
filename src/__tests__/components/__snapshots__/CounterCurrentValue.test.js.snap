// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CounterCurrentValue /> should matches the stored snapshot 1`] = `"Counter current value :  10 "`;
